test_runner/src/main/kotlin/ftl/gc/GcStorage.kt

```diff
@@ -79,6 +80,24 @@ object GcStorage {
         }
     }
 
+    fun uploadCiJUnitXml(testResult: JUnitTestResult, args: IArgs) {
+        if (args.resultsBucket.isEmpty() || args.resultsDir.isNullOrEmpty()) return
+        val progress = ProgressBar()
+        try {
+            progress.start("Uploading JUnit for CI")
+            upload(
+                file = args.ciJUnitResultFile!!,
+                fileBytes = testResult.xmlToString().toByteArray(),
+                rootGcsBucket = args.resultsBucket,
+                runGcsPath = args.resultsDir!!
```

Comment:
Is there a better way to handle this without force unwraps? I think the error message might be confusing if the value ends up being null.

Reply:
> To maximize compatibility, I think we should default to writing `JUnitReport.xml` with the flaky failure nodes removed. Then a flag, `full-junit-result: false` which defaults to off can be added. If full junit results is set, then we'd produce an additional report `JUnitReport_full.xml` that includes the full history of the test executions (including failure nodes for flaky tests that ran multiple times and passed on retry).

But what with merged file generated on gcs? It's generated outside flank by default. When we add this option we will be have locally one file without fail nodes but on gcs we will have two files (merged and cleared). It will be fine behaviour?